proc/process.c

```diff
@@ -809,6 +924,13 @@ static void process_exec(thread_t *current, process_spawn_t *spawn)
 		hal_spinlockClear(&spawn->sl, &sc);
 	}
 
+	if (current->process->tls.tls_base != NULL) {
+		err = tls_init(&current->tls, &current->process->tls, current->process->mapp);
+		if (err == EOK) {
+			hal_cpuTlsSet(&current->tls, current->context);
```

Comment:
What will happen if user code that expects TLS to be present gets a process without TLS? I think it will crash or malfunction in a bad way (i.e. variables we expect to be thread safe are not). I think exec should fail if tls_base != NULL and tls_init failed.

Reply:
I think the check `if (err < 0) {` lower, in line 934 covers tls_init failure